src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py

```diff
@@ -121,66 +154,107 @@ def bulk_record(
         if db_write_keys.size == 0:
             return db_read_key_results
 
-        config = get_ingest_config(use_case_id, IndexerStorage.POSTGRES)
+        config = get_ingest_config(
+            self._get_metric_path_key(strings.keys()), IndexerStorage.POSTGRES
+        )
```

Comment:
Why do you need to fetch the right config based on the use case contained in the metrics you are processing? 
That seems to be error prone. A given indexer instance should always pick up the same configuration at startup and never change it. So it should be more straightforward than this. 

While this approach may work as well, it is problematic in terms of design which makes things less maintainable going on:
- logical dependencies. An indexer instance is supposed to always pick up the same config set up at the start. For somebody who reads the code with less context on it, this code says the opposite. It says that depending on the input, the indexer can pick up a different config each time. Which is incorrect. This discrepancy makes things more confusing.
- robustness. This approach essentially changes the design from: "one indexer instance per config" to "arbitrary number of indexers - each one can pick the right config depending on the input metrics". This would not necessarily be a bad idea, though it does not really work because, if you have multiple use cases in the `strings` variable that require different configs, you raise an Exception.

Reply:
That's fair, the behaviour of obtaining the config at each bulk_record call is a behaviour inherited from the previous revision, and I also see your point that before the `UseCaseKey` (AKA `MetricPathKey`) is passed in through config while in this iteration we are inferring it from the `UseCaseID`s.

Since the `UseCaseKey` (AKA `MetricPathKey`) is static throughout the lifetime of this instance, how would you feel if we set it to a member when it's created here?

https://github.com/getsentry/sentry/blob/john/use-case-aware-indexers-incremental-2/src/sentry/sentry_metrics/consumers/indexer/processing.py#L35-L36

Would this also address your comments for L206?

Reply:
@fpacifici @john-z-yang it seems like we have to get the config in order to pass it to the `writes_limiter` down below, but maybe the indexer should be initialized with the config?